genesis_notification/dm/models.py

```diff
@@ -0,0 +1,309 @@
+        with smtplib.SMTP(self.host, self.port) as smtp:
+            return smtp.sendmail(
+                from_addr=self.noreply_email_address,
+                to_addrs=user_context["email"],
+                msg=msg.as_string(),
+            )
+
+
+class Provider(
+    models.ModelWithUUID,
+    models.ModelWithRequiredNameDesc,
+    ModelWithAlwaysActiveStatus,
+    models.ModelWithProject,
+    models.ModelWithTimestamp,
+    orm.SQLStorableWithJSONFieldsMixin,
+):
+    __tablename__ = "proveders"
+    __jsonfields__ = ["protocol"]
+
+    protocol = properties.property(
+        types_dynamic.KindModelSelectorType(
+            types_dynamic.KindModelType(SimpleSmtpProtocol),
+        ),
+        required=True,
+    )
+
+    def send(self, message, user_context):
+        return self.protocol.send(
+            message=message,
+            user_context=user_context,
+        )
+
+
+class EventType(
+    models.ModelWithUUID,
+    models.ModelWithNameDesc,
+    ModelWithAlwaysActiveStatus,
+    models.ModelWithProject,
+    models.ModelWithTimestamp,
+    orm.SQLStorableMixin,
+):
+    __tablename__ = "event_types"
+
+
+class Template(
+    models.ModelWithUUID,
+    models.ModelWithRequiredNameDesc,
+    ModelWithAlwaysActiveStatus,
+    models.ModelWithProject,
+    models.ModelWithTimestamp,
+    orm.SQLStorableWithJSONFieldsMixin,
+):
+    __tablename__ = "templates"
+    __jsonfields__ = ["params"]
+
+    content = properties.property(
+        types.String(max_length=10240),
+        required=True,
+    )
+    params = properties.property(
+        types.Dict(),
+        required=True,
+    )
+    provider = relationships.relationship(
+        Provider,
+        required=True,
+        prefetch=True,
+    )
+    event_type = relationships.relationship(EventType, required=True)
+    is_default = properties.property(
+        types.Boolean(),
+        default=False,
+    )
+
+
+class UserExcenge(types_dynamic.AbstractKindModel):
+    KIND = "User"
+
+    user_id = properties.property(
+        types.UUID(),
+        required=True,
+    )
+
+
+class ProjectExcenge(types_dynamic.AbstractKindModel):
+    KIND = "Project"
+
+    project_id = properties.property(
+        types.UUID(),
+        required=True,
+    )
+
+
+class SystemExcenge(types_dynamic.AbstractKindModel):
+    KIND = "System"
+
+
+class Binding(
+    models.ModelWithUUID,
+    models.ModelWithProject,
+    ModelWithAlwaysActiveStatus,
+    models.ModelWithTimestamp,
+    orm.SQLStorableMixin,
+):
+    user = properties.property(
+        types.UUID(),
+        required=True,
+    )
+    template = relationships.relationship(
+        Template,
+        required=True,
+    )
+    event_type = relationships.relationship(
+        EventType,
+        required=True,
+    )
+
+
+class Event(
+    models.ModelWithUUID,
+    models.ModelWithNameDesc,
+    models.ModelWithProject,
+    models.ModelWithTimestamp,
+    orm.SQLStorableWithJSONFieldsMixin,
```

Comment:
We can try to use https://github.com/infraguys/restalchemy/pull/41 , after that PR you won't need to add SQLStorableWithJSONFieldsMixin and __jsonfields__ .